### PR TITLE
Set permanent and substitute targets independently

### PR DESCRIPTION
**Pull request by guiccbr**
The permanent target and the substitute target are now sent independently to
the vehicle according to the position of the models gzsitl_perm_targ_ctrl and
gzsitl_subs_targ_ctrl, respectively. We have also included two new models,
gzsitl_perm_targ_vis and gzsitl_subs_targ_vis, that have their positions
updated according to the real mission and detour waypoints currently set on the
vehicle.

Auto-rotate has also been removed from gzsitl. It will be made available in the
mavlink_vehicles instead.

Signed-off-by: Guilherme Campos Camargo guilherme.campos.camargo@intel.com

### CODE REVIEW

File: gzsitl/gzsitl_plugin.cc [hunk 1]
```diff
@@ -120,37 +119,52 @@ void GZSitlPlugin::OnUpdate()
...
-        this->subs_target->SetWorldPose(target_pose);
+    this->subs_target_exists = (bool)this->subs_target;
+
+    // Retrieve current substitute target pose if target exists
+    gazebo::math::Pose subs_target_pose = gazebo::math::Pose::Zero;
+    if (this->subs_target_exists) {
+        subs_target_pose = this->subs_target->GetWorldPose();
+    } else if(is_target_overridden()) {
+        subs_target_pose = this->get_subs_target_pose();
+
```
Comment (dakerfp) [inline gzsitl/gzsitl_plugin.cc:141]: Blank line

File: gzsitl/gzsitl_plugin.cc [hunk 2]
```diff
@@ -120,37 +119,52 @@ void GZSitlPlugin::OnUpdate()
...
     this->subs_target = model->GetWorld()->GetModel(subs_target_name);
-    if (this->subs_target) {
-        this->subs_target->SetWorldPose(target_pose);
+    this->subs_target_exists = (bool)this->subs_target;
+
+    // Retrieve current substitute target pose if target exists
+    gazebo::math::Pose subs_target_pose = gazebo::math::Pose::Zero;
+    if (this->subs_target_exists) {
+        subs_target_pose = this->subs_target->GetWorldPose();
+    } else if(is_target_overridden()) {
```
Comment (dakerfp) [inline gzsitl/gzsitl_plugin.cc:139]: space

File: gzsitl/gzsitl_plugin.cc [hunk 3]
```diff
@@ -120,37 +119,52 @@ void GZSitlPlugin::OnUpdate()
...
-    // Calculate target relative pose if exists
-    gazebo::math::Pose rel_target_pose;
-    if (this->target_exists) {
-        rel_target_pose = target_pose - vehicle_pose;
+    // Update permanent target visualization according to the vehicle
+    mavlink_vehicles::local_pos perm_targ_pos =
+        mavlink_vehicles::math::global_to_local_ned(
+            this->mav->get_mission_waypoint(),
+            this->mav->get_home_position_int());
+    if (this->perm_target_vis =
```
Comment (dakerfp) [inline gzsitl/gzsitl_plugin.cc:149]: this attribution is too long to fit nicely inside the if.

File: gzsitl/gzsitl_plugin.cc [hunk 4]
```diff
@@ -459,3 +386,22 @@ bool GZSitlPlugin::is_target_overridden()
                .count() < defaults::GZSITL_SUBS_TARG_POSE_SUB_MAX_RESPONSE_TIME;
 }
 
+gazebo::math::Vector3 GZSitlPlugin::gazebo_local_to_global(gazebo::math::Pose p)
+{
+
```
Comment (dakerfp) [inline gzsitl/gzsitl_plugin.cc]: Here too

File: .gitmodules [hunk 5]
```diff
@@ -1,3 +1,3 @@
 [submodule "modules/mavlink_vehicles"]
 	path = modules/mavlink_vehicles
-	url = https://github.com/01org/mavlink-vehicles.git
+	url = https://github.com/guiccbr/mavlink-vehicles.git
```
Comment (dakerfp) [inline .gitmodules]: Should this address really be changed?

Comment (guiccbr) [reply .gitmodules]: Just temporarily. I expect [this patch](https://github.com/01org/mavlink-vehicles/pull/5) to be merged before this one. When that happens, I'll get rid of this commit and get back to the original address.

Comment (dakerfp) [reply .gitmodules]: I would prefer to let that happen before changing this temporarily.

Comment (guiccbr) [reply .gitmodules]: Sure. I won't merge this commit before that happens.

**Comment by guiccbr** [comment]
All fixes made. Merging.

**Pull request closed without merging**